### ISSUE TITLE
В каталоге src нужно чистить устаревшие каталоги, у которых нет соответствия реальным epf-файлам

### ISSUE THREAD

**Issue by artbear** [xDrivenDevelopment/precommit1c#28]
В каталоге src нужно чистить устаревшие каталоги, у которых нет соответствия реальным epf-файлам

**Comment by artbear** [comment]
@nixel2007 Сможешь решить эту проблему?

**Comment by nixel2007** [comment]
Если честно, не совсем понимаю, зачем.

**Comment by artbear** [comment]
Примеры
1 при первом коммите у тебя бинарник имел название `Файл1`, в каталоге исходников появится каталог с тем же именем `Файл1` с нужными исходниками.
На втором коммите ты понимаешь, что название лучше поменять и даешь название `Файл2`, делаешь коммит, в каталоге исходников появляется каталог `Файл2` со всем продублированным содержимым
Но при этом каталог `Файл1` не удаляется, т.к. Гит про него ничего не знает
Переименования исходных файлов также может не быть, т.к. у тебя дубль файлов, а не заменя названия.

В итоге для одного файла у тебя может быть 2 каталога с продублированным содержимым :(

2 Также может быть вариант, когда бинарник уже удалили из проекта, а каталог исходников остался :(

Например, для проекта xUnitFor1C 
+ есть несколько похожих наборов [src/Tests/CommonApp/ГенерацияМакетаДанных_На_БазеРеальныхДанных](https://github.com/xDrivenDevelopment/xUnitFor1C/tree/develop/src/Tests/CommonApp/%D0%93%D0%B5%D0%BD%D0%B5%D1%80%D0%B0%D1%86%D0%B8%D1%8F%D0%9C%D0%B0%D0%BA%D0%B5%D1%82%D0%B0%D0%94%D0%B0%D0%BD%D0%BD%D1%8B%D1%85_%D0%9D%D0%B0_%D0%91%D0%B0%D0%B7%D0%B5%D0%A0%D0%B5%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D1%85%D0%94%D0%B0%D0%BD%D0%BD%D1%8B%D1%85) и [src/Tests/CommonApp/тесты_ГенерацияМакетаДанных_На_БазеРеальныхДанных](https://github.com/xDrivenDevelopment/xUnitFor1C/tree/develop/src/Tests/CommonApp/%D1%82%D0%B5%D1%81%D1%82%D1%8B_%D0%93%D0%B5%D0%BD%D0%B5%D1%80%D0%B0%D1%86%D0%B8%D1%8F%D0%9C%D0%B0%D0%BA%D0%B5%D1%82%D0%B0%D0%94%D0%B0%D0%BD%D0%BD%D1%8B%D1%85_%D0%9D%D0%B0_%D0%91%D0%B0%D0%B7%D0%B5%D0%A0%D0%B5%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D1%85%D0%94%D0%B0%D0%BD%D0%BD%D1%8B%D1%85)
+ или пример, когда нет бинарника [src/Tests/CommonApp/тесты_ОткрытиеФормОтчетовОбработок](https://github.com/xDrivenDevelopment/xUnitFor1C/tree/develop/src/Tests/CommonApp/%D1%82%D0%B5%D1%81%D1%82%D1%8B_%D0%9E%D1%82%D0%BA%D1%80%D1%8B%D1%82%D0%B8%D0%B5%D0%A4%D0%BE%D1%80%D0%BC%D0%9E%D1%82%D1%87%D0%B5%D1%82%D0%BE%D0%B2%D0%9E%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BE%D0%BA)

**Comment by nixel2007** [comment]
1. Если переименование делать через git mv, то в диффе гита будет про это написано, эту инфу можно распарсить и соответственно переименовать исходники. Если переименование делали средствами DE/обычным `mv`, то в диффе будет строка с удалением и добавлением нового файла.
2. Эта ситуация тоже будет отражаться в диффе как удаление файла.

Мое предложение - анализировать дифф на предмет именно *удаления* и *переименования* обработок. При удалении соответственно удалять старые каталоги. При корректном переименовании через `git mv` можно так же через `git mv` переименовывать и папку - так мы еще и историю исходников сохраним.

**Comment by artbear** [comment]
@nixel2007 Согласен, так лучше
Сможешь доработать?

**Comment by nixel2007** [comment]
Да, думаю, смогу. Это все будет работать только в режиме `git-precommit`, правильно понимаю?

**Comment by artbear** [comment]
Да, конечно
